core/src/main/scala/com/salesforce/op/stages/impl/feature/TimePeriodTransformer.scala

```diff
@@ -0,0 +1,41 @@
+package com.salesforce.op.stages.impl.feature
+
+import com.salesforce.op.UID
+import com.salesforce.op.features.types.{Date, Integral}
+import com.salesforce.op.stages.base.unary.UnaryTransformer
+import com.salesforce.op.utils.date.DateTimeUtils
+import org.joda.time.{DateTime => JDateTime}
+import com.salesforce.op.features.types._
+
+import scala.reflect.runtime.universe.TypeTag
+
+/**
+ * TimePeriodTransformer extracts one of a set of time periods from a date/datetime
+ *
+ * @param period        time period to extract from date
+ * @param uid           uid for instance
+ * @param tti           type tag for input
+ * @tparam I            input feature type
+ */
+class TimePeriodTransformer[I <: Date]
+(
+  val period: TimePeriod,
+  uid: String = UID[TimePeriodTransformer[_]]
+)(
+  implicit override val tti: TypeTag[I]
+) extends UnaryTransformer[I, Integral](operationName = "dateToTimePeriod", uid = uid){
+  val periodFun: Long => Int = period match {
```

Comment:
just make this a regular function `def periodConversion(time: Long): Int = period match { ... }`